vivadoプロジェクトの自動生成makefileの動作確認

ogakihiromu opened the pull request:
### vivadoプロジェクトの自動生成makefile作ったから、動かしてみて

- 動かし方：cd jt12/vivadoして、あとはmakeするだけ
- 正常動作：guiでjt12のプロジェクトが立ち上がって、simulationでfm_sendの正弦波が見れたらok
- 未確認箇所：jt12/vivado/script/board_inst.tclのボード情報ファイルのgit clone実行部分  
fpgaの型番をボード情報得るファイルをgit cloneするスクリプト  
自分の環境ではすでにボード情報ファイルをvivado側で認識しているため、テストしていない。

ogakihiromu committed:
add makefile and script
    Type make to generate a jt12 project and open it in gui

ogakihiromu committed:
fix .gitignore

nagaolb committed:
Fix openssl error

nagaolb committed:
Fix wrong git command

nagaolb committed:
Fix error when executing git via /bin/sh

nagaolb commented on vivado/script/board_inst.tcl:
ボード設定ファイルが無い際のクローンで
```
/usr/lib/git-core/git-remote-https: /opt/ngon/Xilinx/Vivado/2024.1/tps/lnx64/python-3.8.3/lib/libssl.so.3: version `OPENSSL_3.2.0' not found (required by /usr/lib/libcurl.so.4)
```
のようなエラーが出る現象の修正
……のはずだったのだが、これでも動かない
TCLスクリプト内ではgit関連のコマンドはやめた方がいいのかも
TCLの外にgitを出そうとすると、今度はボードファイルの有無の確認が厄介だが

ogakihiromu committed:
fix makefile and board_inst.tcl

ogakihiromu commented:
tclコンソールの出力をlogファイルとして出力し、
tclコンソール外でlogファイルを読み分岐、gitを使うようにした

ogakihiromu committed:
fix board_inst.tcl

nagaolb committed:
Fix the issue where make command board overlaps directory name

ogakihiromu committed:
fix make
 Changes to be committed:
	modified:   vivado/makefile
	modified:   vivado/script/board_inst.tcl
	modified:   vivado/ym2203_project_regen.tcl
